src/backend/distributed/commands/create_citus_local_table.c

```diff
@@ -0,0 +1,620 @@
+	 */
+	EnsureDependenciesExistOnAllNodes(&tableAddress);
+
+	CreateCitusLocalTable(relationId);
+
+	relation_close(relation, NoLock);
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * ErrorIfUnsupportedCreateCitusLocalTable errors out if we cannot create the
+ * citus local table from the relation relationId.
+ */
+static void
+ErrorIfUnsupportedCreateCitusLocalTable(Oid relationId)
+{
+	/* citus local tables can only be created from coordinator for now */
+	EnsureCoordinator();
+
+	EnsureTableOwner(relationId);
+
+	/* we allow creating citus local tables only from relations with RELKIND_RELATION */
+	EnsureRelationKindSupported(relationId);
+
+	if (PartitionTable(relationId) || PartitionedTable(relationId))
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("citus local tables can not be involved in a "
+							   "partition relationship")));
+	}
+
+	EnsureTableNotDistributed(relationId);
+
+	if (!CoordinatorAddedAsWorkerNode())
+	{
+		const char *relationName = get_rel_name(relationId);
+
+		Assert(relationName != NULL);
+
+		ereport(ERROR, (errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
+						errmsg(
+							"cannot create citus local table \"%s\", citus local "
+							"tables can only be created from coordinator node if "
+							"it is added to pg_dist_node", relationName)));
+	}
+}
+
+
+/*
+ * CreateCitusLocalTable is the internal method to create citus local table's
+ * shard and metadata.
+ * Note that this function does not perform any validation on the table as it
+ * is already done in create_citus_local_table.
+ */
+static void
+CreateCitusLocalTable(Oid shellRelationId)
+{
+	/*
+	 * Make sure that existing reference tables have been replicated to all
+	 * the nodes such that we can create foreign keys and joins work
+	 * immediately after creation.
+	 */
+	EnsureReferenceTablesExistOnAllNodes();
+
+	/*
+	 * Get an exclusive lock on relation with shellRelationId as the operations
+	 * done in this function should be automic.
+	 */
+	LockRelationOid(shellRelationId, ExclusiveLock);
+
+	/*
+	 * Get necessary commands to recreate the shell table before renaming the
+	 * given relation to the shard relation.
+	 */
+	List *foreignConstraintCommands =
+		GetForeignConstraintCommandsTableInvolved(shellRelationId);
+
+	/*
+	 * Include DEFAULT clauses for columns getting their default values from
+	 * a sequence.
+	 */
+	bool includeSequenceDefaults = true;
+
+	List *shellTableDDLEvents = GetTableDDLEvents(shellRelationId,
+												  includeSequenceDefaults);
+	shellTableDDLEvents = list_concat(shellTableDDLEvents, foreignConstraintCommands);
+
+	uint64 shardId = GetNextShardId();
+
+	char *shellRelationName = get_rel_name(shellRelationId);
+	Oid shellRelationSchemaId = get_rel_namespace(shellRelationId);
+
+	/* below we convert relation with shellRelationId to the shard relation */
+	CreateCitusLocalTableShard(shellRelationId, shardId);
+
+	/*
+	 * As we retrieved the DDL commands necessary to create the shell table
```

Comment:
Actually a similar comment is actually written a few lines above:

```c
/* below we convert relation with shellRelationId to the shard relation */
```

. Putting a similar comment to here might make the comments a bit repetitive ?

Reply:
Actually here we are not recreating the shell table logically, because there wasn't a shell table, there was a regular postgres table. It might make things more clear to apply this comment on  this PR https://github.com/citusdata/citus/pull/3852#discussion_r433816508, does it make sense?